test/test_torch.py

```diff
@@ -17532,6 +17532,19 @@ def test_pickle_gradscaler(self, device):
                 if lazy_init_scale:
                     self.assertEqual(b.scale(torch.tensor([4.0], dtype=torch.float32, device=device)), 12.0)
 
+    @dtypesIfCUDA(torch.half, torch.float, torch.double,
```

Comment:
This should also test a complex dtype, either to ensure they work (if nansum supports them) or that they throw a runtime error. That way if someone adds complex support for this op they'll know to update the test.

Reply:
Makes sense.

Reply:
Done.